### PR TITLE
Python2->Python3 update,   Python2 unsafe

### PR DESCRIPTION
**Pull request by qinzhang22**
For python-pam, this is already built with xs8 and xs9. The XS8 package is in rt-next now.

xsconsole xenrt test passed: 3864890

### CODE REVIEW

**Review by liulinC — APPROVED** [review]

**Comment by liulinC** [comment]
Do not forget to tag before this merge, with **major version bump up** (for in-compatibility)

File: plugins-base/XSFeatureSRCommon.py [hunk 1]
```diff
@@ -226,7 +226,7 @@ def UpdateFieldsCONFIRM(self):
         pane.ResetFields()
 
         sr = HotAccessor().sr[self.srHandle]
-        srName = sr.name_label(None).encode('utf-8')
+        srName = Lang(sr.name_label(None))
```
Comment (rosslagerwall) [inline plugins-base/XSFeatureSRCommon.py:229]: Why've you used `Lang()` here and elsewhere in the commit? I think it is not normally used with dynamic values. Normally it is only used with static labels like `Operation`, `Storage Repository`, etc.

Comment (qinzhang22) [reply plugins-base/XSFeatureSRCommon.py:229]: I used `Lang()` as I'm not sure the object is a string or a bytes object. Let's remove `encode` here to see how it goes.

**Review by GeraldEV — APPROVED** [review]

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]
Please apply change request that I originally made in PR 18 to keep Python3 working:
https://github.com/xapi-project/xsconsole/pull/18#discussion_r1380876618

Here is Pau's hard disagree comment again:
https://github.com/xapi-project/xsconsole/pull/15#issuecomment-1780887363
Andrew's initiative to keep Python2
https://github.com/xapi-project/xsconsole/pull/15#issuecomment-1781785654
and also Alex questioned splitting the code base of packages recently when there is no need to do so.


This change is small and even reduces the amount of changed lines:
```py
try:
    import xmlrpclib
    import SocketServer
    import SimpleXMLRPCServer
except:
    import xmlrpc.client as xmlrpclib
    import socketserver as SocketServer
    import xmlrpc.server as SimpleXMLRPCServer
```

In addition, there are xsconsole plugins on xs8. You can find one with `rpm -qa on xs8`
```sh
rpm -qa | grep xsconsole
```

What about xsconsole plugins by partners?
Can we check them?

Until these are checked and updated as well, we cannot declare xsconsole update work as finished.

**Comment by qinzhang22** [comment]
> Please apply change request that I originally made in PR 18 to keep Python3 working: [#18 (comment)](https://github.com/xapi-project/xsconsole/pull/18#discussion_r1380876618)
> 
> Here is Pau's hard disagree comment again: [#15 (comment)](https://github.com/xapi-project/xsconsole/pull/15#issuecomment-1780887363) Andrew's initiative to keep Python2 [#15 (comment)](https://github.com/xapi-project/xsconsole/pull/15#issuecomment-1781785654) and also Alex questioned splitting the code base of packages recently when there is no need to do so.
> 
> This change is small and even reduces the amount of changed lines:
> 
> ```python
> try:
>     import xmlrpclib
>     import SocketServer
>     import SimpleXMLRPCServer
> except:
>     import xmlrpc.client as xmlrpclib
>     import socketserver as SocketServer
>     import xmlrpc.server as SimpleXMLRPCServer
> ```
> 
> In addition, there are xsconsole plugins on xs8. You can find one with `rpm -qa on xs8`
> 
> ```shell
> rpm -qa | grep xsconsole
> ```
> 
> What about xsconsole plugins by partners? Can we check them?
> 
> Until these are checked and updated as well, we cannot declare xsconsole update work as finished.

We has a new agreement here: https://github.com/xapi-project/xsconsole/pull/18#issuecomment-1791844449. There's no dependency of xsconsole that's going to need it to stay py2 compatible. And my PR is working on XS8. So what's your point we have to make such change here?

**Comment by bernhardkaindl** [comment]
Merged in split form with #24, #25, #28 and the change force the use of Python3 is no longer needed.

**Pull request closed without merging**